### ISSUE TITLE
Attach to local process by name?

### ISSUE THREAD

**Issue by etotheipi** [jmxtrans/jmxtrans#494]
Happy to do a PR to add this functionality, if desired.

For security reasons, we are trying to avoid exposing a jmxremote port for local process monitoring.  As far as I can tell, the only other way is by PID.  However, if jmxtrans relies on a static JSON file and we want this deployment automated, it seems we have to start the process to be monitored first, then fetch the PID and then rewrite the JSON file using the new PID.

I'm wondering if there's an easier way to do this, already.  If not, I'd be happy to contribute code to do this.  The `jps` utility makes it fairly easy to find these local processes by name or CLI arguments, and I think it would be straightforward for jmxtrans to use this information instead of PID or host:port.  

I understand the risk of multiple JVMs matching a given description, but there will always be a way for the developer to uniquely identify which JVM they want (perhaps by embedding a unique string the CLI args).  We can choose to just get the "first" reported JVM that is a match (and thus undefined behavior if there are multiple matches), or produce and error.

**Comment by gehel** [comment]
No, there is no way to attach to local process by name in jmxtrans. If you have a PR ready, I'd be happy to merge it!

**Comment by gehel** [comment]
Actually, you might want to have a look at [jmxtrans-agent](https://github.com/jmxtrans/jmxtrans-agent) which might be a better fit for this specific use case.
